unity-client/Assets/Scripts/MainScripts/DCL/UserProfile/UserProfileController.cs

```diff
@@ -30,32 +33,26 @@ public void Awake()
 
     public void LoadProfile(string payload)
     {
-        if (payload == null)
+        if (!baseWearablesAlreadyRequested)
         {
-            return;
+            baseWearablesAlreadyRequested = true;
+            CatalogController.RequestBaseWearables()
+                             .Catch((error) =>
+                             {
+                                 OnBaseWereablesFail?.Invoke();
+                                 Debug.LogError(error);
+                             });
         }
 
-        var model = JsonUtility.FromJson<UserProfileModel>(payload);
+        if (payload == null)
+            return;
 
-        CatalogController.RequestBaseWearables()
-            .Then((baseWearables) =>
-            {
-                CatalogController.RequestOwnedWearables(model.userId)
-                    .Then((ownedWearables) =>
-                    {
-                        ownUserProfile.SetInventory(ownedWearables.Select(x => x.id).ToArray());
-                        ownUserProfile.UpdateData(model);
-                        userProfilesCatalog.Add(model.userId, ownUserProfile);
-                    })
-                    .Catch((error) => Debug.LogError(error));
-            })
-            .Catch((error) => Debug.LogError(error));
+        var model = JsonUtility.FromJson<UserProfileModel>(payload);
```

Comment:
could we wrap the json parsing inside a try/catch? just to be safe :)

Reply:
In this case I don't see the need to wrap the parsing because if it fails, wouldn't provoke any block in the main flow. Notice that this `LoadProfile()` function is called in a isolated way only when we receive the `LoadProfile` message from kernel and it is only updating the user profile. In case this parsing fails, the profile simply doesn't be updated, we would see the corresponding exception in our console, but the main flow of the application would continue.